components/sup/src/manager/service/config.rs

```diff
@@ -174,11 +175,20 @@ impl Cfg {
             .replace("-", "_");
         match env::var(&var_name) {
             Ok(config) => {
-                let toml = toml::de::from_str(&config).map_err(|e| {
-                    sup_error!(Error::TomlParser(e))
-                })?;
-                self.environment = Some(toml::Value::Table(toml));
-
+                match toml::de::from_str(&config) {
```

Comment:
The "short circuit"/"early return" pattern I was thinking of would make this function look a bit like this:

```rust
fn load_environment(&mut self, package: &Pkg) -> Result<()> {
    let var_name = format!("{}_{}", ENV_VAR_PREFIX, package.name)
        .to_ascii_uppercase()
        .replace("-", "_");
    match env::var(&var_name) {
        Ok(config) => {
            match toml::de::from_str(&config) {
                Ok(toml) => {
                    self.environment = Some(toml::Value::Table(toml));
                    return Ok(());
                },
                Err(err) => debug!("Attempted to parse env config as toml and failed, {}", err),
            }
            match serde_json::from_str(&config) {
                Ok(json) => {
                    self.environment = Some(toml::Value::Table(json));
                    return Ok(());
                },
                Err(err) => debug!("Attempted to parse env config as json and failed, {}", err),
            }
            self.environment = None;
            Err(Error::BadEnvConfig(package.name.to_ascii_uppercase()))
        }
        Err(e) => {
            debug!(
                "Looking up environment variable {} failed: {:?}",
                var_name,
                e
            );
            self.environment = None;
            Ok(())
        }
    }
}
```

I recommend this pattern for this function becuase it let's you "fall through" and then eventually fail